### PR TITLE
Implement solution 20 for project euler

### PR DESCRIPTION
**Pull request by G1Joshi**
# Welcome to Dart community

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_20/sol20.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  for (int i = factorial(number); i > 0; i = (i / 10).floor()) {
+    sum += (i % 10);
+  }
+  return sum;
+}
+
+// Driver Code
+void main() {
+  int number = 100;
+  print(sum(number));
```
Comment (akashgk) [inline project_euler/problem_20/sol20.dart:31]: add test cases

**Review by akashgk — CHANGES_REQUESTED** [review]
LGTM! Please add some tests

**Comment by G1Joshi** [comment]
> LGTM! Please add some tests

Thank you :)
I have added 3 test cases.

File: project_euler/problem_20/sol20.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  return sum;
+}
+
+// Driver Code
+void main() {
+  group("Solution 20", () {
+    test("Test 1", () {
+      var matcher = 1;
+      var actual = sum(1);
```
Comment (akashgk) [inline project_euler/problem_20/sol20.dart:35]: no need for variable declarations

File: project_euler/problem_20/sol20.dart [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      expect(actual, matcher);
+    });
+
+    test("Test 2", () {
+      var matcher = 3;
+      var actual = sum(5);
+      expect(actual, matcher);
```
Comment (akashgk) [inline project_euler/problem_20/sol20.dart:42]: same

File: project_euler/problem_20/sol20.dart [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    test("Test 3", () {
+      var matcher = 27;
+      var actual = sum(10);
```
Comment (akashgk) [inline project_euler/problem_20/sol20.dart:47]: no need for variable declaration.

**Review by akashgk — CHANGES_REQUESTED** [review]
all is good. but we don't need to declare the variables for each case. just give the values directly to the function

Also, do a dart format on your file.

**Comment by G1Joshi** [comment]
> all is good. but we don't need to declare the variables for each case. just give the values directly to the function
> 
> Also, do a dart format on your file.

Test cases are updated and the code is already formatted.

**Comment by akashgk** [comment]
Thanks @G1Joshi

**Pull request merged**